common-pg/src/main/scala/com/socrata/pg/soql/SqlFunctions.scala

```diff
@@ -538,7 +538,7 @@ object SqlFunctions extends SqlFunctionsLocation with SqlFunctionsGeometry with
                  ctx: Sqlizer.Context,
                  escape: Escape): ParametricSql = {
 
-    val templateTypeCast = if (fn.window.isEmpty && typeCastIfNotWindowFn.isDefined) template + typeCastIfNotWindowFn.map("::" + _).get
+    val templateTypeCast = if (fn.window.isEmpty && fn.filter.isEmpty && typeCastIfNotWindowFn.isDefined) template + typeCastIfNotWindowFn.map("::" + _).get
```

Comment:
so this is now like... typecast if not window or filter function?

Reply:
type if (not window) and (not filter) and ...

Reply:
i was handling count(column) but missed count(*) because count_star is a weird function that is handled differently.

Reply:
just the naming has become misleading is what I'm observing: `typeCastIfNotWindowFn`

Reply:
But I don't know what a better name would be without getting really silly-long

Reply:
That part isn't changed although the name is kinda tied.  I can rename that to typeCastIfNotWindowFnAndNotFilter or simply typeCast.